aea/helpers/base.py

```diff
@@ -802,12 +811,41 @@ def public_key_or_identifier(self) -> str:
 
     def get_message(self, public_key: str) -> bytes:  # pylint: disable=no-self-use
         """Get the message to sign."""
-        message = public_key.encode("ascii")
-        # + self.identifier.encode("ascii")  # noqa: E800
-        # + self.not_before_string.encode("ascii")  # noqa: E800
-        # + self.not_after_string.encode("ascii")  # noqa: E800
+        message = self.construct_message(
+            public_key,
+            self.identifier,
+            self.not_before_string,
+            self.not_after_string,
+            self.message_format,
+        )
         return message
 
+    @classmethod
+    def construct_message(
+        cls,
+        public_key: str,
+        identifier: SimpleIdOrStr,
+        not_before_string: str,
+        not_after_string: str,
+        message_format: str,
+    ) -> bytes:
+        """
+        Construct message for singning.
+
+        :param public_key: the public key
+        :param identifier: identifier to be signed
+        :param not_before_string: signature not valid before
+        :param not_after_string: signature not valid after
+        :param message_format: message format used for signing
+        """
+        message = message_format.format(
```

Comment:
as specification, this could also be ordered based, rather than 'key' based (although in that case the message format is less readable)

Reply:
what's the value in that?

Reply:
shorter format messages, in cases we wanted to put an upperbound